### PR TITLE
Adds BlockDevices for LaunchConfigurations

### PR DESCRIPTION
**Pull request by jschneiderhan**
Adds Block Device support for Launch Configurations.

The code is mostly copied from the EC2 package and tests.

### CODE REVIEW

**Comment by jschneiderhan** [comment]
Oops - I actually think I need to change BlockDevices to BlockDeviceMappings in a few places.

**Comment by statik** [comment]
:+1: I definitely need this for a project I'm working on. I hope it can be reviewed and merged soon. Anything I can do to help or test?

**Comment by jschneiderhan** [comment]
@statik just waiting for review by a maintainer. It seems to work when paired with https://github.com/hashicorp/terraform/pull/935. I've never written a line of Go before this, so if you have a min to do a quick review it would be helpful to avoid any back-and-forth due to stupid errors when the maintainers get a chance to look at it :smile:.

File: autoscaling/autoscaling.go [hunk 1]
```diff
@@ -298,6 +355,7 @@ func (autoscaling *AutoScaling) CreateLaunchConfiguration(options *CreateLaunchC
 		b64.Encode(userData, []byte(options.UserData))
 		params["UserData"] = string(userData)
 	}
+	addBlockDeviceParams("", params, options.BlockDevices)
```
Comment (phinze) [inline autoscaling/autoscaling.go]: If you're always passing an empty string for the first arg, should we just drop the `prename` arg?

**Comment by phinze** [comment]
Hey there - thanks for this! One quick question inline but otherwise looking good!

File: autoscaling/autoscaling.go (hunk 1 above)
Comment (mitchellh) [inline autoscaling/autoscaling.go]: I'm guessing he just copied the function from another package. Its fine, we'll switch to aws-go soon enough anyways.

**Comment by mitchellh** [comment]
LGTM.

File: autoscaling/autoscaling.go (hunk 1 above)
Comment (jschneiderhan) [inline autoscaling/autoscaling.go]: correct - copied from the ec2 package.

**Pull request merged**